core/src/saros/negotiation/AbstractIncomingProjectNegotiation.java

```diff
@@ -502,25 +514,26 @@ protected void deleteCollectors() {
     startActivityQueuingRequestCollector.cancel();
   }
 
-  protected void checkProjectMapping(final Map<String, IProject> mapping) {
+  protected void checkProjectMapping(final Map<String, IReferencePoint> mapping) {
 
-    for (final Entry<String, IProject> entry : mapping.entrySet()) {
+    for (final Entry<String, IReferencePoint> entry : mapping.entrySet()) {
 
       final String id = entry.getKey();
-      final IProject project = entry.getValue();
+      final IReferencePoint referencePoint = entry.getValue();
 
       final ProjectNegotiationData data = getProjectNegotiationData(id);
 
-      if (data == null) throw new IllegalArgumentException("invalid project id: " + id);
+      if (data == null) throw new IllegalArgumentException("invalid reference point id: " + id);
 
-      if (!project.exists())
-        throw new IllegalArgumentException("project does not exist: " + project);
+      if (!referencePointManager.projectExists(referencePoint))
+        throw new IllegalArgumentException("reference point does not exist: " + referencePoint);
```

Comment:
`throw new IllegalArgumentException("project does not exist at the referencePoint " + referencePoint);`

Reply:
done